pkg/internal/apis/config/validate_test.go

```diff
@@ -428,3 +431,141 @@ func TestPortValidate(t *testing.T) {
 		})
 	}
 }
+
+func TestValidatePortMapping(t *testing.T) {
+	newPortMapping := func(addr string, port int, protocol string) PortMapping {
+		return PortMapping{
+			HostPort:      int32(port),
+			ListenAddress: addr,
+			Protocol:      PortMappingProtocol(protocol),
+		}
+	}
+	errMsg := "port mapping with same listen address, port and protocol already configured"
+	cases := []struct {
+		testName     string
+		portMappings []PortMapping
+		expectErr    string
+	}{
+		{
+			testName: "unique port mappings ipv4",
+			portMappings: []PortMapping{
+				newPortMapping("127.0.0.1", 80, "UDP"),
+				newPortMapping("127.0.0.1", 80, "TCP"),
+				newPortMapping("0.0.0.0", 3000, "UDP"),
+				newPortMapping("0.0.0.0", 5000, "TCP"),
+			},
+			expectErr: "",
+		},
+		{
+			testName: "unique port mappings ipv6",
+			portMappings: []PortMapping{
+				newPortMapping("::1", 80, "UDP"),
+				newPortMapping("::1", 80, "TCP"),
+				newPortMapping("1e3d:6e85:424d:a011:a72e:9780:5f6f:a6fc", 3000, "UDP"),
+				newPortMapping("6516:944d:e070:a1d1:2e91:8437:a6b3:edf9", 5000, "TCP"),
+			},
+			expectErr: "",
+		},
+		{
+			testName: "unique port mappings ipv4 & ipv6",
+			portMappings: []PortMapping{
+				newPortMapping("127.0.0.1", 80, "TCP"),
+				newPortMapping("0.0.0.0", 80, "UDP"),
+				newPortMapping("::1", 80, "TCP"),
+				newPortMapping("::", 80, "UDP"),
```

Comment:
this should fail , in golang `::` and `0.0.0.0` are equivalent, https://github.com/golang/go/issues/48723